client/src/lib/data.ts

```diff
@@ -225,14 +234,79 @@ export async function readCalendar(calendarId: string): Promise<Calendar> {
   return calendar;
 }
 
-export type Mark = {
-  markId: number;
+type CreateCalParams = {
+  type: string;
+  name: string;
+  color: string;
+  desc?: string;
+  goal: number;
+};
+export async function createCal({
+  type,
+  name,
+  color,
+  desc,
+  goal,
+}: CreateCalParams): Promise<Calendar> {
+  const req = {
+    body: JSON.stringify({
```

Comment:
If you just name the parameter (say, `cal`) then you can simply write `JSON.stringify(cal)`. Then you wouldn't need to destructure it and potentially have a bug if you modify the Calendar fields.